src/extensions/base-extension.js

```diff
@@ -0,0 +1,315 @@
+    }
+  }
+
+  setExtensionPathInScope(scopePath: string): void {
+    const componentPath = _getExtensionPath(this.name, scopePath, this.options.core);
+    this.filePath = componentPath;
+  }
+
+  static transformStringToModelObject(name: string) {
+    return {
+      name,
+      config: {}
+    };
+  }
+
+  /**
+   * Load extension by name
+   * The extension will be from scope by default or from file
+   * if there is file(path) in the options
+   * The file path is relative to the bit.json of the project or absolute
+   * @param {string} name - name of the extension
+   * @param {Object} rawConfig - raw config for the extension
+   * @param {Object} options - extension options such as - disabled, file, core
+   * @param {string} consumerPath - path to the consumer folder (to load the file relatively)
+   * @param {string} scopePath - scope which stores the extension code
+   */
+  static async load({
+    name,
+    rawConfig = {},
+    options = {},
+    consumerPath,
+    scopePath
+  }: BaseLoadArgsProps): Promise<BaseExtensionProps> {
+    // logger.info(`loading extension ${name}`);
+    // Require extension from _debugFile
+    const concreteBaseAPI = _getConcreteBaseAPI({ name });
+    if (options.file) {
+      let absPath = options.file;
+      if (!path.isAbsolute(options.file) && consumerPath) {
+        absPath = path.resolve(consumerPath, options.file);
+      }
+      const staticExtensionProps: StaticProps = await BaseExtension.loadFromFile(name, absPath, rawConfig, options);
+      const extensionProps: BaseExtensionProps = { api: concreteBaseAPI, ...staticExtensionProps };
+      extensionProps.api = concreteBaseAPI;
+      return extensionProps;
+    }
+    let staticExtensionProps: StaticProps = {
+      name,
+      rawConfig,
+      dynamicConfig: rawConfig,
+      options,
+      disabled: false,
+      loaded: false,
+      filePath: ''
+    };
+    // Require extension from scope
+    if (scopePath) {
+      const componentPath = _getExtensionPath(name, scopePath, options.core);
+      staticExtensionProps = await BaseExtension.loadFromFile(name, componentPath, rawConfig, options);
+    }
+    const extensionProps: BaseExtensionProps = { api: concreteBaseAPI, ...staticExtensionProps };
+    return extensionProps;
+  }
+
+  static loadFromModelObject(modelObject) {
+    let staticExtensionProps: StaticProps;
+    if (typeof modelObject === 'string') {
+      staticExtensionProps = {
+        name: modelObject,
+        rawConfig: {},
+        dynamicConfig: {},
+        options: {},
+        disabled: false,
+        loaded: false,
+        filePath: ''
+      };
+    } else {
+      staticExtensionProps = {
+        name: modelObject.name,
+        rawConfig: modelObject.config,
+        dynamicConfig: modelObject.config,
+        options: {},
+        disabled: false,
+        loaded: false,
+        filePath: ''
+      };
+    }
+
+    const concreteBaseAPI = _getConcreteBaseAPI({ name: modelObject.name });
```

Comment:
maybe you want to do `{ name: staticExtensionProps.name }` instead. seems like `modelObject` may be a string